src/session.ts

```diff
@@ -348,52 +349,76 @@ export class Session extends DebugSession {
             return;
         }
 
-        let originalLocalAbsolutePath = path.normalize(args.source.path);
+        // store source breakpoints per file
+        this._sourceBreakpointsMap.set(args.source.path, args.breakpoints);
 
-        const originalBreakpoints = args.breakpoints || [];
-        const generatedBreakpoints: DebugProtocol.SourceBreakpoint[] = [];
-        let generatedRemoteLocalPath = undefined;
+        // rebuild the generated breakpoints map each time a breakpoint is changed in any file
+        let generatedBreakpointsMap: Map<string, DebugProtocol.SourceBreakpoint[]> = new Map();
 
-        try {
-            // first get generated remote file path, will throw if fails
-            generatedRemoteLocalPath = await this._sourceMaps.getGeneratedRemoteRelativePath(originalLocalAbsolutePath);
-
-            // for all breakpoint positions set on the source file, get generated/mapped positions
-            if (originalBreakpoints.length) {
-                for (let originalBreakpoint of originalBreakpoints) {
-                    const generatedPosition = await this._sourceMaps.getGeneratedPositionFor({
-                        source: originalLocalAbsolutePath,
-                        column: originalBreakpoint.column || 0,
-                        line: originalBreakpoint.line,
-                    });
-                    generatedBreakpoints.push({
-                        line: generatedPosition.line || 0,
-                        column: 0,
-                    });
+        // get generated breakpoints from all sources
+        for (let [sourcePath, sourceBreakpoints] of this._sourceBreakpointsMap) {
+            let originalLocalAbsolutePath = path.normalize(sourcePath);
+
+            const originalBreakpoints = sourceBreakpoints || [];
+            let generatedRemoteLocalPath = undefined;
+
+            try {
+                // first get generated remote file path, will throw if fails
+                generatedRemoteLocalPath = await this._sourceMaps.getGeneratedRemoteRelativePath(
+                    originalLocalAbsolutePath
+                );
+
+                // append to any existing breakpoints for this generated file
+                if (!generatedBreakpointsMap.has(generatedRemoteLocalPath)) {
+                    generatedBreakpointsMap.set(generatedRemoteLocalPath, []);
+                }
+                const generatedBreakpoints = generatedBreakpointsMap.get(generatedRemoteLocalPath)!;
+
+                // for all breakpoint positions set on the source file, get generated/mapped positions
+                if (originalBreakpoints.length) {
+                    for (let originalBreakpoint of originalBreakpoints) {
+                        const generatedPosition = await this._sourceMaps.getGeneratedPositionFor({
+                            source: originalLocalAbsolutePath,
+                            column: originalBreakpoint.column || 0,
```

Comment:
Using null coalescing operator should be better:
```suggestion
                            column: originalBreakpoint.column ?? 0,
```

Reply:
Thanks for the tip. Looks like I have this in a LOT of places.